pkg/policy/resolve_test.go

```diff
@@ -44,14 +46,30 @@ var (
 
 func (ds *PolicyTestSuite) SetUpSuite(c *C) {
 	SetPolicyEnabled(option.DefaultEnforcement)
-	repo.AddList(GenerateNumRules(1000000))
-
+	GenerateNumIdentities(3000)
+	repo.AddList(GenerateNumRules(1000))
 }
 
 func (ds *PolicyTestSuite) TearDownSuite(c *C) {
 	repo = &Repository{}
 }
 
+func GenerateNumIdentities(numIdentities int) {
+	for i := 0; i < numIdentities; i++ {
+
+		identityLabel := labels.NewLabel(fmt.Sprintf("k8s:foo%d", i), "", "")
+
+		identityLabels := labels.Labels{
+			fmt.Sprintf("foo%d", i): identityLabel,
```

Comment:
This is a very short label compared to what we typically. I think it would make sense to add about 5 labels with actual key-value pairs to get it closer to reality in terms of cost in comparing strings. You will also see a much bigger gain with caching. You could use the namespace and serviceaccount examples to make it quite realistic or just use the kube-dns labels from one of the well-known identities and add an additional ID labelt o make it unique.